File: 1º Semestre/Programação de Computadores/Dev C++ exercícios/Lista de exercício 03/11 - while.cpp
/* Elabore um algoritmo que leia um número de entrada que indicará a quantidade de
números a serem lidos. Em seguida, leia n números (conforme o valor informado
anteriormente) e imprima a soma e a média aritmética dos números informados. (while)*/ 

#include <stdio.h>

int main(){
	int numero, contador = 1, qtdade, soma = 0;
	float media;
	
	
	printf("Informe a quantidade de entradas de numeros para operacoes de soma e media aritmetica: ");
	scanf("%d", &qtdade);
	
	printf("Informe o 1o numero: ");
	scanf("%d", &numero);
	soma = soma + numero;
	contador++;
	
	while(contador <= qtdade){
		printf("Informe o %do numero: ", contador);
		scanf("%d", &numero);
		soma = soma + numero;
		media = (float)soma / qtdade;
		contador++;
	}
	
	printf("A soma dos %d numeros informados e %d.\n", qtdade, soma);
	printf("A media aritmetica dos %d numeros informados e de %.2f.\n", qtdade, media); 
	
	return 0;
}

File: 1º Semestre/Programação Python/Listas de Exercícios/Exercício 3/11 - while.py
"""11. Elabore um algoritmo que leia um número de entrada que indicará a quantidade de
números a serem lidos. Em seguida, leia n números (conforme o valor informado
anteriormente) e imprima a soma e a média aritmética dos números informados."""

contador = 1
soma = 0

qtdade = int(input("Informe o número que servirá para a quantidade de números das operações de soma e média "
                   "aritmética: "))

num = int(input("Informe o 1o número: "))
soma = soma + num
contador += 1

while contador <= qtdade:
    num = int(input(f"Informe o {contador}o número: "))
    soma = soma + num
    media = soma/num
    contador += 1

print(f"A soma dos números entre 1 e {num} é: {soma}")
print(f"A média aritmética dos números entre 1 e {num} é: {media:.2f}")
